Controller/ProductViewController.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\ProductBundle\Controller;
+
+use Sulu\Bundle\ProductBundle\Entity\ProductInterface;
+use Sulu\Bundle\ProductBundle\Entity\ProductTranslation;
+use Sulu\Bundle\ProductBundle\Product\ProductFactoryInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * This controller is used for viewing product-templates.
+ */
+class ProductViewController extends Controller
+{
+    /**
+     * This action is used for displaying a product via a template.
+     * The template that is defined by sulu_product.template parameter is used for displaying
+     * the product.
+     *
+     * @param ProductInterface $product
+     * @param ProductTranslation $translation
+     *
+     * @return Response
+     */
+    public function viewAction(ProductInterface $product, ProductTranslation $translation)
+    {
+        $apiProduct = $this->getProductFactory()->createApiEntity($product, $translation->getLocale());
+
+        return $this->render(
+            $this->getProductViewTemplate(),
+            [
+                'product' => $apiProduct,
```

Comment:
`urls` variable should also be passen which is a assoziative array from locale to url